WcaOnRails/app/models/competition.rb

```diff
@@ -98,11 +98,15 @@ class Competition < ActiveRecord::Base
 
   validate :must_have_at_least_one_event, if: :confirmed_or_visible?
   private def must_have_at_least_one_event
-    if competition_events.reject(&:marked_for_destruction?).empty?
+    if no_events
       errors.add(:competition_events, I18n.t('competitions.errors.must_contain_event'))
     end
   end
 
+  def no_events
+    return competition_events.reject(&:marked_for_destruction?).empty?
```

Comment:
Also, the ruby convention for methods that return a boolean value is to always end their name with a question mark, so `no_events` -> `no_events?`.